### ISSUE TITLE
sortable table does not track prop changes

### ISSUE THREAD

**Issue by rdy** [pivotal-cf/pivotal-ui#172]
a sortable table loads the content from the props into the state, but when the props change, it should probably re-read the changed props and apply the sort. instead does not handle the new props.

this is a problem when dynamically loading content for a sortable table, it would be nice if the table would track the props, and apply the sort when they change.

**Comment by stubbornella** [comment]
@rdy - Thanks for the bug report. Do you mean when you add a new column or a new row to the table?

**Comment by rdy** [comment]
@stubbornella we tried it specifically with changing rows, but I would expect it to work the same way if you happened to add columns it should also track those new columns and data correctly.

**Comment by gpleiss** [comment]
Based on a conversation with @rdy, it sounds like a good fix to this problem would be to redo the whole interface to this component. Redoing the interface should fix this issue

The problem with the current interface is that there is no way to customize the innards of the table.

![screen shot 2015-07-07 at 10 20 40 am](https://cloud.githubusercontent.com/assets/824157/8552420/d187b4fa-2491-11e5-9014-0ca53063e9da.png)

Say we wanted to add a custom class to all the "names", or we wanted to add a click handler on the "synergy" column. With the current implementation, there is no way to do this.

A better interface would look like:

```jsx
var headers = [
  <Header sortable={true}>Name</Header>
  <Header>Email Address</Header>
];

<SortableTable headers={headers}>
  {data.map(function(datum) {
    return (
      <TableRow>
        <Td>{datum.name}</Td>
        <Td>{datum.email}</Td>
      </TableRow>
    );
  })}
</SortableTable>
```

This makes the component much more React-y, and consistent with our other components. (This component has a weird interface because it was the first component we made :grin:). Also, changing this interface would solve the issue above.

**Comment by ctaymor** [comment]
This should be fixed in 2.0.0-alpha.
